infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationService.kt

```diff
@@ -707,6 +700,41 @@ class PublicationService @Autowired constructor(
         return fieldErrors + referenceErrors + switchErrorsSegments + switchErrorsTopological + duplicateErrors + alignmentErrors + geocodingErrors + duplicateNameErrors
     }
 
+    private fun validateDuplicateOf(
+        version: ValidationVersion<LocationTrack>,
+        locationTrack: LocationTrack,
+        validationVersions: ValidationVersions,
+    ): List<PublishValidationError> {
+        val duplicatesAfterPublication = locationTrackDao.fetchDuplicates(version.officialId)
+            .map(locationTrackDao::fetch)
+            .mapNotNull { duplicate -> fetchRelevantTrackVersionForDuplicateCheck(validationVersions, duplicate) }
+            .filter { potentialDuplicate -> potentialDuplicate.duplicateOf == version.officialId }
+
+        val duplicateOf = locationTrack.duplicateOf?.let { duplicateId ->
+            getLocationTrack(duplicateId, validationVersions)
+        }
+        return validateDuplicateOfState(
+            locationTrack,
+            duplicateOf,
+            validationVersions.locationTracks.map { it.officialId },
+            duplicatesAfterPublication,
+        )
+    }
+
+    private fun fetchRelevantTrackVersionForDuplicateCheck(
+        validationVersions: ValidationVersions,
+        track: LocationTrack,
+    ): LocationTrack? {
+        val trackIsInPublicationUnit =
+            validationVersions.locationTracks.any { validationVersion -> validationVersion.officialId == track.id }
+
+        return if (trackIsInPublicationUnit && track.getDraftType() == DraftType.OFFICIAL)
```

Comment:
Ensinnäkin, eikös nuo versiot haettu niin että sinne tulee sekä draft että official? Silloin näkisin että:
- Jos joku draft-versio on duplikaatti mutta se ei ole julkaisujoukossa mukana, se voidaan sivuuttaa (vain official on kiinnostava ja jos se on duplikaatti niin sen pitäisi olla löydetyssä versiojoukossa myös)
- Jos joku official-versio on duplikaatti ja se on versiojoukossa mukana, se voidaan vastaavasti sivuuttaa sillä tuon draftin pitäisi myös olla mukana jos se on edelleen duplikaatti

Toiseksi, jos tuo raide on julkaisussa mukana, olisi parempi käyttää julkaisun versiota, ei hakea uutta draft-versiota, sillä julkaisussa kiinnitetään aluksi validaatio-versiot ja juuri ne tulee julkaistuksi. Jos väliin kerkiää muutos, se päätyy romukoppaan. Tämä on tietenkin hiukan ärsyttävää tarkastaa tuota versiota kunnioittaen (eikä se ihan kaikissa muissakaan funkkareissa täällä mene oikein), mutta tässä minusta hoituisi ihan suhteellisen nätisti, jotenkin näin:
- Hae kaikki official duplikaattiversiot (draftilla on väliä vain jos se on validointijoukossa)
- Lisää päälle kaikki validaatiojoukon versiot (duplikaatti tai ei), ylikirjoittaen ne officialit jotka on molemmissa
- Hae noiden versioiden oliot
- Filtteröi ne jotka ovat duplikaatteja (tämähän tehdään jo nyt tuolla kutsupaikassa)

Tuossa pitäisi olla mukana kaikki mikä voi lopputilassa olla duplikaattia, ilman että tarvitsee tupla-hakea mitään versioita tai tarkastella tiloja kuten täällä.